app/routes/bdb/components/CompanyEditor.tsx

```diff
@@ -13,28 +14,40 @@ import LegoFinalForm from 'app/components/Form/LegoFinalForm';
 import SubmissionError from 'app/components/Form/SubmissionError';
 import { SubmitButton } from 'app/components/Form/SubmitButton';
 import InfoBubble from 'app/components/InfoBubble';
-import type {
-  CompanyEntity,
-  SubmitCompanyEntity,
-} from 'app/reducers/companies';
+import type { CompanyEntity } from 'app/reducers/companies';
+import type { AutocompleteEntity } from 'app/store/models/Autocomplete';
+import type { AdminDetailCompany } from 'app/store/models/Company';
 import { createValidator, required, isEmail } from 'app/utils/validation';
 import { httpCheck, DetailNavigation, ListNavigation } from '../utils';
 import styles from './bdb.css';
+import type { EntityId } from '@reduxjs/toolkit';
 
 type Props = {
   uploadFile: (arg0: Record<string, any>) => Promise<any>;
   company: CompanyEntity;
   autoFocus: any;
   fetching: boolean;
-  submitFunction: (arg0: SubmitCompanyEntity) => Promise<any>;
-  deleteCompany: (arg0: number) => Promise<any>;
+  submitFunction: (company: AdminDetailCompany) => Promise<any>;
+  deleteCompany: (id: EntityId) => Promise<any>;
 };
 
 const validate = createValidator({
   name: [required()],
   paymentMail: [isEmail()],
 });
 
+type FormValues = {
+  logo?: string;
+  name?: string;
+  description?: string;
+  companyType?: string;
+  website?: string;
+  address?: string;
+  studentContact?: AutocompleteEntity<AutocompleteContentType.User>;
+};
+
+const TypedLegoForm = LegoFinalForm<FormValues>;
```

Comment:
I don't like that this name is so generic, when it's typed just to fit this form. A bit nitpicky, but still

Reply:
I think it makes sense like this. The file it is in gives context about what form it is typed to fit, just like `Props`.